nilearn/glm/contrasts.py

```diff
@@ -412,32 +475,88 @@ def compute_fixed_effects(contrast_imgs, variance_imgs, mask=None,
         masker = NiftiMasker(mask_img=mask).fit()
 
     variances = masker.transform(variance_imgs)
-    contrasts = masker.transform(contrast_imgs)
+    contrasts = np.array(
+        [masker.transform(contrast_img) for contrast_img in contrast_imgs]
+    )
+
+    if dofs is not None:
+        if len(dofs) != n_runs:
+            raise ValueError(
+                f"The number of degrees of freedom ({len(dofs)}) "
+                f"differs from the number of contrast images ({n_runs})."
+            )
+    else:
+        dofs = [100] * n_runs
 
-    (fixed_fx_contrast,
-     fixed_fx_variance, fixed_fx_t) = _compute_fixed_effects_params(
-        contrasts, variances, precision_weighted)
+    (
+        fixed_fx_contrast,
+        fixed_fx_variance,
+        fixed_fx_stat,
+        fixed_fx_z_score,
+    ) = _compute_fixed_effects_params(
+        contrasts, variances, precision_weighted, dofs
+    )
 
     fixed_fx_contrast_img = masker.inverse_transform(fixed_fx_contrast)
     fixed_fx_variance_img = masker.inverse_transform(fixed_fx_variance)
-    fixed_fx_t_img = masker.inverse_transform(fixed_fx_t)
-    return fixed_fx_contrast_img, fixed_fx_variance_img, fixed_fx_t_img
+    fixed_fx_stat_img = masker.inverse_transform(fixed_fx_stat)
+    fixed_fx_z_score_img = masker.inverse_transform(fixed_fx_z_score)
+    warn(
+        category=FutureWarning,
+        message="The default behavior of the function will "
+        "be changed and replaced in release 0.13 "
+        "to have an additional return value 'fixed_fz_z_score_img'  "
+        "by default",
```

Comment:
```suggestion
        "by default.",
```

Do we want to send the warning all the time or only when `return_z_score` is False?